src/librustc_mir/dataflow/impls/maybe_init_locals.rs

```diff
@@ -0,0 +1,126 @@
+pub use super::*;
+
+use crate::dataflow::{BitDenotation, GenKillSet};
+use rustc::mir::visit::{
+    MutatingUseContext, NonMutatingUseContext, NonUseContext, PlaceContext, Visitor,
+};
+use rustc::mir::*;
+
+/// This calculates if any part of a MIR local may still be initialized.
+/// This means that once a local has been written to, its bit will be set
+/// from that point and onwards, until we see an Operand::Move out of, or
+/// StorageDead statement for the local.
+/// This is an approximation of `MaybeInitializedPlaces` for whole locals.
+#[derive(Copy, Clone)]
+pub struct MaybeInitializedLocals<'a, 'tcx> {
+    body: &'a Body<'tcx>,
+}
+
+impl<'a, 'tcx> MaybeInitializedLocals<'a, 'tcx> {
+    pub fn new(body: &'a Body<'tcx>) -> Self {
+        MaybeInitializedLocals { body }
+    }
+
+    pub fn body(&self) -> &Body<'tcx> {
+        self.body
+    }
+}
+
+impl<'a, 'tcx> BitDenotation<'tcx> for MaybeInitializedLocals<'a, 'tcx> {
+    type Idx = Local;
+    fn name() -> &'static str {
+        "has_been_borrowed_locals"
```

Comment:
This should get changed

Reply:
Oops! You can tell I started by copy-pasting lol.